ldap/servers/slapd/dyncerts.c

```diff
@@ -798,6 +797,786 @@ dyncerts_cert2entry(CERTCertificate *cert)
     return e;
 }
 
+/* Free data alloced by nss_add_cert_and_key within CertCtx_t*/
+static void
+dyncerts_free_nss_add_cert_and_key(CertCtx_t *ctx)
+{
+    if (ctx->cert) {
+        CERT_DestroyCertificate(ctx->cert);
+        ctx->cert = NULL;
+    }
+    if (ctx->privkey) {
+        if (ctx->ldaprc) {
+            PK11_DeleteTokenPrivateKey(ctx->privkey, PR_FALSE);
+        } else {
+            SECKEY_DestroyPrivateKey(ctx->privkey);
+        }
+        ctx->privkey = NULL;
+    }
+}
+
+
+/* Free data within CertCtx_t */
+void
+dyncerts_cert_ctx_done(CertCtx_t *ctx)
+{
+    if (ctx->dercert.data) {
+         slapi_ch_free((void**)&ctx->dercert.data);
+         ctx->dercert.data = NULL;
+    }
+    if (ctx->derpkey.data) {
+        slapi_ch_free((void**)&ctx->derpkey.data);
+        ctx->derpkey.data = NULL;
+    }
+    dyncert_nickname_free(&ctx->n);
+    slapi_ch_free_string(&ctx->trust);
```

Comment:
This fields is sometime set with static string (e.g. "CT,,") we should have a flag to free it or not